src/orchard/namespace.clj

```diff
@@ -29,12 +36,26 @@
                     (catch Exception _
                       ::fail))]
         (cond
-          (#{::eof ::fail} found) nil
-          (and (list? found)
-               (-> found first #{`ns 'ns})
-               (symbol? (second found)))
-          (second found)
-          :else (recur))))))
+          (#{::eof ::fail} found)
+          nil
+
+          (pred found)
+          (extract found)
+
+          :else
+          (recur))))))
+
+(defn read-namespace
```

Comment:
Perhaps `ns-name` instead? `read-namespace` could also be interpreted as `ns-form`. (the function below)

Reply:
It's its existing name, but we can deprecate it in favor of that name?

Reply:
What an existing name? `read-ns-name` or `ns-name`?

Reply:
Ah, `read-namespace` was moved. Yeah, I guess it can be renamed and the old name can be deprecated.

Reply:
Done!